rusty_pizza_server/src/money.rs

```diff
@@ -0,0 +1,383 @@
+        // When:
+        let _ = Money::new(7, 20) - Money::new(7, 40);
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u8() {
+        // When:
+        let result = Money::new(5, 0) * 2u8;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u8_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u8;
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn u8_can_be_multiplied_with_money() {
+        // When:
+        let result = 2u8 * Money::new(5, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn u8_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u8 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u16() {
+        // When:
+        let result = Money::new(5, 0) * 2u16;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u16_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u16;
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn u16_can_be_multiplied_with_money() {
+        // When:
+        let result = 2u16 * Money::new(5, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn u16_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u16 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u32() {
+        // When:
+        let result = Money::new(5, 0) * 2u32;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u32_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u32;
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn u32_can_be_multiplied_with_money() {
+        // When:
+        let result = 2u32 * Money::new(5, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn u32_can_be_multiplied_with_money_with_different_values() {
+        // When:
+        let result = 3u32 * Money::new(2, 0);
+
+        // Then:
+        assert_eq!(result, Money { cents: 600 })
+    }
+
+    #[test]
+    fn money_is_copy() {
+        // Given:
+        let money = Money::new(2, 0);
+
+        // When:
+        let result1 = 3u32 * money;
+        let result2 = money + Money::new(1, 0);
+
+        // Then:
+        assert_eq!(result1, Money { cents: 600 });
+        assert_eq!(result2, Money { cents: 300 });
```

Comment:
Test if the original money values is available

Reply:
Done.